client/js/widgets/widget.js

```diff
@@ -1497,16 +1497,22 @@ export class Widget extends StateManaged {
           if(typeof nextTurnIndex != 'number' || !isFinite(nextTurnIndex))
             nextTurnIndex = 0;
           const turn = indexList[mod(nextTurnIndex, indexList.length)];
+          variables.TURN = turn;
 
           collections[a.collection] = [];
-          //saves turn into all seats and creates output collection with turn seats
-          for(const w of c) {
-            await w.set('turn', w.get('index') == turn);
-            if(w.get('turn') && w.get('player'))
-              collections[a.collection].push(w);
+
+          if(!a.readOnly) {
+            //saves turn into all seats and creates output collection with turn seats
+            for(const w of c) {
+              await w.set('turn', w.get('index') == turn);
+            }
+            jeLoggingRoutineOperationSummary(`changed turn of seats from ${previousTurn} to ${turn} - active seats: ${JSON.stringify(indexList)}`);
+          } else {
+            jeLoggingRoutineOperationSummary(`SIMULATED turn of seats: from ${previousTurn} to ${turn} - active seats: ${JSON.stringify(indexList)}`);
           }
 
-          jeLoggingRoutineOperationSummary(`changed turn of seats from ${previousTurn} to ${turn} - active seats: ${JSON.stringify(indexList)}`);
+          collections[a.collection] = c.filter(w=>w.get('index')==turn && w.get('player'));
+
```

Comment:
Can't this stay inside the `!a.readOnly` branch? If it's readonly, none of these things have changed, so the collection will remain unchanged.

Reply:
@ArnoldSmith86?